code/modules/GOAP/monkey/info_provider.dm

```diff
@@ -0,0 +1,72 @@
+/datum/goap_info_provider/monkey/GetWorldState(datum/goap_agent/agent)
+	. = list()
+	.["GUN"] = FALSE
+	.["attackEnemy"] = FALSE
+	.["disposeEnemy"] = FALSE
+	.["Flee"] = FALSE
+	.["disarmAvailable"] = FALSE
+	.["enemyGrabbed"] = FALSE
+	.["replaceItem"] = FALSE
+	var/mob/living/carbon/monkey/C = agent.agent
+	var/obj/item/W = locate() in C.held_items
+	if(W)
+		.["hasItem"] = TRUE
+		.["hasHumanItem"] = TRUE
+		.["grabbedWeapon"] = TRUE
+		if(isgun(W))
+			var/obj/item/gun/G = W
+			if(G && G.chambered && G.chambered.BB) // we got a loaded gun!
+				.["GUN"] = TRUE
+	else
+		C.best_force = 0
+		.["hasItem"] = FALSE
+		.["hasHumanItem"] = FALSE
+		.["grabbedWeapon"] = FALSE
+	if(C.pulling)
+		.["enemyGrabbed"] = TRUE
+	var/mob/living/carbon/human/H = locate(/mob/living/carbon/human/) in view(4, agent.agent)
+	if(H && H.held_items)
+		if(prob(MONKEY_ATTACK_DISARM_PROB)) // this runs more than monkeys handle_combat did
+			.["disarmAvailable"] = TRUE
+
+/datum/goap_info_provider/monkey/GetGoal(datum/goap_agent/agent)
+	. = list()
+	var/mob/living/carbon/monkey/C = agent.agent
+	var/obj/item/W = locate() in C.held_items
+	var/itemobtainable = FALSE
+	if(length(C.enemies))
+		var/list/around = view(MONKEY_ENEMY_VISION, C)
+		for(var/mob/living/L in around)
```

Comment:
ditto GLOB

Reply:
won't work here because the else if looks for disposals in the area